retis/src/collect/collector/ovs/ovs.rs

```diff
@@ -146,9 +146,6 @@ impl Module for OvsModule {
     fn collector(&mut self) -> &mut dyn Collector {
         self
     }
-    fn section_factory(&self) -> Result<Option<Box<dyn EventSectionFactory>>> {
-        Ok(Some(Box::new(OvsEventFactory::new()?)))
```

Comment:
I haven't gone through the whole series so maybe I'm missing something but...

This has a significant side-effect. Factories can no longer depend on anything collector-specific, for example, that depends on user input.

Here is an example (unmerged): https://github.com/amorenoz/retis/blob/da48e9c119b2fe9035f42d2455b653a19856d52b/retis/src/module/ovs/ovs.rs#L170-L174

Here ^ we create a collector-specific component (only if user provided some arguments) and we pass that to the factory so they can interact.

Is this possible with this PR?

Reply:
This is not possible atm but mostly because there was no use case. It could easily be done by accessing the factory using the `SectionFactories` object after the call to `section_factories()`, to set extra options. (This would require casting the factory to its real type, which is a common construction in Retis).

Reply:
As per other comment, we have a way forward. Thanks.